src/pm/util/rm.c

```diff
@@ -374,7 +374,20 @@ MachineTable *MPIE_ReadMachines(const char *arch, int nNeeded, void *data)
         nNeeded--;
     }
     mt->nHosts = nFound;
+
+  fn_exit:
+    fclose(fp);
     return mt;
+
+  fn_fail:
+    if (mt->desc) {
+        MPL_free(mt->desc);
+    }
```

Comment:
won't this cause a potential NULL dereference?

Reply:
You are right.

Reply:
fixed